src/index.js

```diff
@@ -28,38 +29,74 @@ export function attach(env) {
   env.fetch = fetchWrapper(env.fetch);
 }
 
-function init() {
-  if (isReactNative()) {
-    attach(global);
-  } else if (isWorker()) {
-    attach(self);
-  } else if (isWeb()) {
-    attach(window);
-  } else if (isNode()) {
-    attach(global);
-  } else {
+function initialize() {
+  environment = resolveEnvironment();
+  if (!environment) {
     throw new Error('Unsupported environment for fetch-token-intercept');
   }
+
+  attach(environment);
 }
 
+/**
+ * Initializes and configures interceptor
+ * @param config
```

Comment:
What is this config? Document it, or add `@see` if it is documented somewhere else.